testing/dart/image_shader_test.dart

```diff
@@ -0,0 +1,20 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:typed_data';
+import 'dart:ui';
+
+import 'package:litetest/litetest.dart';
+import 'canvas_test.dart' show createImage, testCanvas;
+
+void main() {
+  test('Construct an ImageShader', () async {
+    final Image image = await createImage(50, 50);
+    final ImageShader shader = ImageShader(image, TileMode.clamp, TileMode.clamp, Float64List(16));
+    final Paint paint = Paint()..shader=shader;
+    final Rect rect = Rect.fromLTRB(0, 0, 100, 100);
+    testCanvas((Canvas canvas) => canvas.drawRect(rect, paint));
```

Comment:
Actually, this is testing the safe path. When you use the shader in a paint in a draw call - that is when the image gets wrapped in a GPUObject. The path that might cause a problem would be:

- create an Image
- create an ImageShader from that image
- Don't use that shader in a rendering call
- drop the Dart reference to the image (so that the flutter::Image reference goes away)
- wait for the unref thread to unref the image that was held from the Image object
- now drop the Shader reference as well
- wait for GC which should try to reclaim the flutter::ImageShader which will unref the SkImage without protection.

Reply:
yes - this test is just an attempt to invoke `ImageShader` in order to verify that a PR like this has not broken the Dart API.

Dart does not offer the fine grained control over GC and engine threads needed for the above scenario.  And the unit test environment will be using a software rasterizer, so it would not show the potential resource leaks that might be seen when deleting the `SkImage` on the UI thread on the Android/OpenGL platform.